include/mbedtls/check_config.h

```diff
@@ -550,6 +555,10 @@
 #error "MBEDTLS_PSA_CRYPTO_C defined, but not all prerequisites (missing RNG)"
 #endif
 
+#if defined(MBEDTLS_PSA_CRYPTO_C) && !defined(MBEDTLS_CIPHER_C )
+#error "MBEDTLS_PSA_CRYPTO_C defined, but not all prerequisites (missing CIPHER)"
```

Comment:
Hmm, the thing is, our implementation depends on `CIPHER_C` (currently), but we allow pluging in drivers, and if they cover all ciphers/aeads, we shouldn't need `CIPHER_C`. So, I'm not sure what's the best way to express this. @gilles-peskine-arm wdyt?

Reply:
At least first function in `psa_crypto_cipher.c` file: `mbedtls_cipher_info_from_psa` calls `mbedtls_cipher_info_from_values`.

Reply:
Oh, indeed that dependency seems to be unconditional and present whether we have drivers or not. Good point. I withdraw this comment and am marking the conversation as resolved.

Reply:
Eventually PSA shouldn't depend on `MBEDTLS_CIPHER_C`, only on individual cipher-related modules for things that aren't done via a driver. (If `MBEDTLS_GCM_C` needs `MBEDTLS_CIPHER_C`, that's its business.) But we're not there yet. A lot of the PSA cipher code depends on the cipher module, both for metadata (which may be needed even when dispatching to drivers) and for operations.